core/src/main/java/com/kin/ecosystem/core/data/blockchain/BlockchainSourceImpl.java

```diff
@@ -98,20 +97,27 @@ public static BlockchainSourceImpl getInstance() {
 		return instance;
 	}
 
+	@Override
+	public void createAccount() throws BlockchainException {
+		createKinAccountIfNeeded();
+		initBalance();
+	}
+
 	private void createKinAccountIfNeeded() throws BlockchainException {
-		int accountIndex = local.getAccountIndex();
 		if (kinClient.hasAccount()) {
-			account = kinClient.getAccount(accountIndex);
+			if (account == null) {
+				account = kinClient.getAccount(activeAccountIndex);
+			}
 		} else {
```

Comment:
instead of else if (account == null){//create new accout}?

Reply:
add TODO:

Reply:
check not -1

Reply:
opened a task